addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -767,339 +1183,339 @@ def test_assert_created_backorders_computed_correctly(self):
+        apple_qty = 10
+        apple_qty_per_line = apple_qty / 2
+
+        # Create quants for apples in two separate packages
+        pack1 = self.create_package()
+        pack2 = self.create_package()
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, apple_qty_per_line, package_id=pack1.id
+        )
+        self.create_quant(
+            self.apple.id, self.test_stock_location_01.id, apple_qty_per_line, package_id=pack2.id
+        )
+
+        # Create picking with demand for all apples in stock, split over 2 lines
+        pick = self.create_picking(
+            self.picking_type_pick,
+            products_info=[
+                {"product": self.apple, "uom_qty": apple_qty_per_line},
+                {"product": self.apple, "uom_qty": apple_qty_per_line},
+            ],
+            location_dest_id=self.test_received_location_01.id,
+            location_id=self.test_stock_location_01.id,
+            assign=True,
+        )
+
+        self.assertEqual(pick.u_quantity_done, 0, "Quantity done should be zero")
+        self.assertEqual(
+            pick.u_total_quantity, apple_qty, "Total quantity should match apple quantity"
+        )
+        self.assertTrue(pick.u_has_discrepancies, "Pick should have discrepancies")
 
-#     def test_unlinks_empty_pickings_if_picking_type_configured(self):
-#         """Empty pickings for a picking type are unlinked if auto-unlinking is enabled."""
-#         Picking = self.env["stock.picking"]
+        # Fulfil 1st move line
+        pick.move_line_ids[0].qty_done = apple_qty_per_line
 
-#         self.picking_type_pick.u_auto_unlink_empty = True
-#         pick = self.create_picking(self.picking_type_pick)
-#         pick.u_is_empty = True
+        self.assertEqual(
+            pick.u_quantity_done,
+            apple_qty_per_line,
+            "Quantity done should match apple per line quantity",
+        )
+        self.assertTrue(pick.u_has_discrepancies, "Pick should have discrepancies")
 
-#         Picking.unlink_empty()
+        # Fulfil 2nd move line which should mean pick is no longer flagged as having discrepancies
+        pick.move_line_ids[1].qty_done = apple_qty_per_line
 
-#         self.assertFalse(pick.exists())
+        self.assertEqual(
+            pick.u_quantity_done, apple_qty, "Quantity done should match apple quantity"
+        )
+        self.assertFalse(pick.u_has_discrepancies, "Pick should not have discrepancies")
 
-#     def test_does_not_unlink_empty_pickings_if_picking_type_not_configured(self):
-#         """Empty pickings for a picking type remain if auto-unlinking is disabled."""
-#         Picking = self.env["stock.picking"]
 
-#         self.picking_type_pick.u_auto_unlink_empty = False
-#         pick = self.create_picking(self.picking_type_pick)
-#         pick.u_is_empty = True
+class TestStockPickingUoM(TestStockPickingCommon):
```

Comment:
We're in a file name "test_stock_picking" so prefixing class names with `TestStockPicking` seems like duplication.  If we remove it it may force us to come up with a better name for the class.